iree/compiler/Dialect/HAL/Target/MetalSPIRV/SPIRVToMSL.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iree/compiler/Dialect/HAL/Target/MetalSPIRV/SPIRVToMSL.h"
+
+#include "llvm/Support/Debug.h"
+#include "llvm/Support/raw_ostream.h"
+#include "spirv_cross/spirv_msl.hpp"
```

Comment:
what's the impact on build time and compiler size? we chatted about this, but linking in even more stuff to the binary is scary :(

Reply:
Agreed with you. But unfortunately making the spirv-cross to run entirely as an offline tool can be hard. We need to set many parameters (like each resource variable's argument index) when generating the MSL, based on SPIR-V reflection. This sort of tasks are best to be done with C++ APIs. For example, see https://github.com/antiagainst/iree/commit/9a841885cd8cea4212a571aa0238db6b520747f6. Actually I'm not sure using the command-line is even possible for all the reflection in that.. This part is for Apple platforms only and it can be excluded if necessary. Overall this compiler target backend is sharing the SPIR-V CodeGen path; the 3 static libraries (`libspirv-cross-core.a`, `libspirv-cross-glsl.a`, `libspirv-cross-msl.a`) we use totals at < 4MB under release.